### DISCUSSION TITLE
Webpack config

### DISCUSSION THREAD

**Discussion by tsxalchemist** [foxglove/community#64] [Q&A]
Hello! I'm a Junior Frontend developer and I'm developing an _extension_ for Studio for my graduate work. As part of my introduction to foxglove, I'm trying to develop from scratch and then refine the extension based on the **Map panel**.

I encountered an error while setting up the environment and compiling the test extension.
After `yarn local-install` I received a message:
![image](https://user-images.githubusercontent.com/66438971/165370069-3bfb64c3-c7d1-4368-9500-6e0008880f74.png)

As I understand, this error is caused by importing _leaflet.css_.
![image](https://user-images.githubusercontent.com/66438971/165370165-cfaba4d8-59c8-4848-afc4-07e757b159e2.png)

Usually when developing in the browser I specify in webpack.config.js loaders to handle CSS and SASS. But in this case I don't quite understand how to do it. I tried just adding the config to the project for webpack, but that didn't work. I suspect it's because @foxglove only uses their configs.

**Question**: how can I add my rules/configs for the webpack so that @foxglove sees them when building? If there is no such way, how can I get around this error?

You can do a `git clone 'https://github.com/Moloko20/gw-extension'` to reproduce the error.

**Comment by jtbandes** [comment]
Hi, and welcome! Could you confirm the version of `fox` that you're using? We added support for CSS to the default webpack config in https://github.com/foxglove/fox/pull/33. The latest version of the fox tool (0.3.0) should include this change.

**Reply by tsxalchemist** [reply]
I am currently using version "0.2.1". I will try to upgrade and come back with feedback here.

**Reply by tsxalchemist** [reply]
Hi! I updated the foxglove dependencies and now I get a new error :(
As I understand, webpack still requires to install some loaders.

Repository: https://github.com/Moloko20/gw-extension

![image](https://user-images.githubusercontent.com/66438971/165584703-503484c9-d94a-4aee-8753-dd71658dfe81.png)

**Reply by tsxalchemist** [reply]
After a while I could figure out that these errors were caused by code I took from the **Map** panel (https://github.com/foxglove/studio/blob/main/packages/studio-base/src/panels/Map/index.tsx).

But how to fix it I did not understand, because without these imports _leaflet_ does not work correctly.

**Code:**
![image](https://user-images.githubusercontent.com/66438971/165777797-12be75fa-9a21-4c93-9713-e46977fe86ee.png)

**Reply by jtbandes** [reply]
I think the problem is now happening because of the `.png` file imports. Created a new feature request for this in fox: https://github.com/foxglove/fox/issues/38

**Reply by jtbandes** [reply]
We also have a separate feature request for ability to customize the webpack config: https://github.com/foxglove/fox/issues/32

**Reply by tsxalchemist** [reply]
@jtbandes Thank you so much, your answers have helped me move forward with the development.

I would really appreciate it if you could approve foxglove/fox#32 as soon as possible and make changes to the main branch.

**Comment by foxymiles** [comment]
https://github.com/foxglove/fox/issues/32 has been merged and v0.4.0 of the package has been released. You should be able to provide your own `config.ts` file now to set up any loaders you need. Please let me know if you run into any issues.

**Reply by tsxalchemist** [reply]
Wow, that was fast! Thank you very much, this will make further development much easier.
